### DISCUSSION TITLE
Help with API methods

### DISCUSSION THREAD

**Discussion by euu2021** [freeplane/freeplane#1542] [General]
I'm testing every single method in the API, in order to build the [Table of API Methods](https://github.com/freeplane/docs/discussions/135). 

I will post here the ones that I'm having difficulty in making work. 
Before posting, I search for examples in the scripts collection, the old forum, Github, and Google.
If anyone can create a working example of any of those methods, please share:

- Attributes	int	`findFirst​(java.lang.String name)`. 
I'm triyng `node.attributes.findFirst​("aaa")`, and it gives me this error:  
![image](https://github.com/freeplane/freeplane/assets/77707706/b399a912-ff1d-48fd-8b69-a71688cdc7c3)
This works fine, though: `node.attributes.containsKey("aaa")`

**Comment by EdoFro** [comment]
here is a working example

[Script samples.zip](https://github.com/freeplane/freeplane/files/13404724/Script.samples.zip)

**Reply by euu2021** [reply]
Thanks. Your example showed that `node.attributes.findFirst​("aaa")` should work. I investigated further and found that, for some reason, my code had an empty character in it:
![image](https://github.com/freeplane/freeplane/assets/77707706/b55e858c-5eb1-44c7-ad0a-e1333d239952)

I deleted it, and now it's working fine.

**Reply by EdoFro** [reply]
that happens when copy/pasting from the API html page.
